src/apps/vpn/addons/state/addonstate.h

```diff
@@ -0,0 +1,52 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef ADDONSTATE_H
+#define ADDONSTATE_H
+
+#include <QJsonObject>
+
+#include "addonsessionstate.h"
+
+/**
+ * @brief Struct containing the different types of state an addon can hold.
+ *
+ * These are:
+ *
+ * - `session` state: A type of state with is not persisted throughout device
+ * sessions.
+ *
+ */
+struct AddonState {
+  Q_GADGET
+  Q_PROPERTY(AddonSessionState* session READ session)
+
+ public:
+  AddonState() = default;
+  ~AddonState() { delete m_session; }
```

Comment:
Why a manual delete? Can you set the parent obj in m_session?

Reply:
Well, because of ignorance really :D Didn't realize that was the reason for setting the parent. Now that you say it, it makes sense :flushed: